sklearn/cross_validation.py

```diff
@@ -1255,108 +1192,11 @@ def cross_val_predict(estimator, X, y=None, cv=None, n_jobs=1,
-    preds_blocks = parallel(delayed(_fit_and_predict)(clone(estimator), X, y,
-                                                      train, test, verbose,
-                                                      fit_params)
-                            for train, test in cv)
-
-    preds = [p for p, _ in preds_blocks]
-    locs = np.concatenate([loc for _, loc in preds_blocks])
-    if not _check_is_partition(locs, _num_samples(X)):
-        raise ValueError('cross_val_predict only works for partitions')
-    inv_locs = np.empty(len(locs), dtype=int)
-    inv_locs[locs] = np.arange(len(locs))
-
-    # Check for sparse predictions
-    if sp.issparse(preds[0]):
-        preds = sp.vstack(preds, format=preds[0].format)
-    else:
-        preds = np.concatenate(preds)
-    return preds[inv_locs]
-
-
-def _fit_and_predict(estimator, X, y, train, test, verbose, fit_params):
-    """Fit estimator and predict values for a given dataset split.
-
-    Read more in the :ref:`User Guide <cross_validation>`.
-
-    Parameters
-    ----------
-    estimator : estimator object implementing 'fit' and 'predict'
-        The object to use to fit the data.
-
-    X : array-like of shape at least 2D
-        The data to fit.
-
-    y : array-like, optional, default: None
-        The target variable to try to predict in the case of
-        supervised learning.
-
-    train : array-like, shape (n_train_samples,)
-        Indices of training samples.
-
-    test : array-like, shape (n_test_samples,)
-        Indices of test samples.
-
-    verbose : integer
-        The verbosity level.
-
-    fit_params : dict or None
-        Parameters that will be passed to ``estimator.fit``.
-
-    Returns
-    -------
-    preds : sequence
-        Result of calling 'estimator.predict'
-
-    test : array-like
-        This is the value of the test parameter
-    """
-    # Adjust length of sample weights
-    fit_params = fit_params if fit_params is not None else {}
-    fit_params = dict([(k, _index_param_value(X, v, train))
-                      for k, v in fit_params.items()])
-
-    X_train, y_train = _safe_split(estimator, X, y, train)
-    X_test, _ = _safe_split(estimator, X, y, test, train)
-
-    if y_train is None:
-        estimator.fit(X_train, **fit_params)
-    else:
-        estimator.fit(X_train, y_train, **fit_params)
-    preds = estimator.predict(X_test)
-    return preds, test
-
-
-def _check_is_partition(locs, n):
-    """Check whether locs is a reordering of the array np.arange(n)
-
-    Parameters
-    ----------
-    locs : ndarray
-        integer array to test
-    n : int
-        number of expected elements
-
-    Returns
-    -------
-    is_partition : bool
-        True iff sorted(locs) is range(n)
-    """
-    if len(locs) != n:
-        return False
-    hit = np.zeros(n, bool)
-    hit[locs] = True
-    if not np.all(hit):
-        return False
-    return True
+    return model_selection.cross_val_predict(
```

Comment:
I did this because the order of the `labels` parameter was changed in the new module... Previously it was `(X, y, cv...)`. At `model_selection`, it is `(X, y, labels...)`...